src/plugins/image/editing/image-editor.js

```diff
@@ -69,24 +69,67 @@ class ImageEditor extends React.Component {
     this.changeContent({ text: newValue });
   }
 
-  handleHoverCopyrightInfoValueChanged(event) {
-    const newValue = event.target.value;
-    const hover = { ...this.props.content.hover };
-    hover.text = newValue;
-    this.changeContent({ hover });
+  handleOrientationValueChanged(e) {
+    const effect = {
+      ...this.props.content.effect, orientation: e.target.value
+    };
+
+    this.changeContent({ effect });
   }
 
-  handleHoverSwitchChange(checked) {
-    if (checked) {
-      this.changeContent({ hover: { type: 'internal', url: '', text: '' } });
-    } else {
-      this.changeContent({ hover: null });
+  handleEffectOptionChange(e) {
+    const baseProps = {
+      sourceType: SOURCE_TYPE.internal,
+      sourceUrl: '',
+      text: ''
+    };
+
+    switch (e.target.value) {
+      case EFFECT_TYPE.reveal:
+        this.changeContent({
+          effect: {
+            ...baseProps,
+            type: EFFECT_TYPE.reveal,
+            startPosition: 10,
+            orientation: ORIENTATION.horizontal
+          }
+        });
+        break;
+      case EFFECT_TYPE.hover:
+        this.changeContent({
+          effect: {
+            ...baseProps,
+            type: EFFECT_TYPE.hover
+          }
+        });
+        break;
+      case EFFECT_TYPE.none:
+        this.changeContent({ effect: null });
+        break;
+      default:
+        this.changeContent({ effect: null });
+
```

Comment:
Did ESLint not require a `break` in this default case? :thinking:

Reply:
nope